### PR TITLE
Adding support for comparison operators.

### PR DESCRIPTION
**Pull request by dhermes**
**NOTE**: This has #21 as diffbase.

### CODE REVIEW

**Comment by dhermes** [comment]
@crwilcox For easier review: https://github.com/dhermes/ndb-rewrite/pull/22/commits/119d6d85a54f920dec9633f7fe447d37bef83855

**Comment by sduskis** [comment]
@dhermes: there are merge conflicts.

File: src/google/cloud/ndb/key.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ class Key:
         Key('Parent', 'C', 'Child', 42)
 
     Either of the above constructor forms can additionally pass in another
-    key using ``parent=<key>``. The ``(kind, id)`` pairs of the parent key are
+    key the ``parent`` keyword. The ``(kind, id)`` pairs of the parent key are
```
Comment (crwilcox) [inline src/google/cloud/ndb/key.py]: Wording here is funny. Suggest perhaps "can additionally pass in an other key, the parent" or "can additionally pass the parent" ?

**Comment by dhermes** [comment]
I'm going to rebase and merge this, which should automagically resolve #21 and #20.

File: src/google/cloud/ndb/key.py (hunk 1 above)
Comment (dhermes) [reply src/google/cloud/ndb/key.py]: Ah shucks I missed this comment before merging. Will send a PR.

**Pull request merged**